src/lib/pipeline/LoadStoreQueue.cc

```diff
@@ -100,6 +104,13 @@ void LoadStoreQueue::startLoad(const std::shared_ptr<Instruction>& insn) {
   if (ld_addresses.size() == 0) {
     // Early execution if not addresses need to be accessed
     insn->execute();
+
+    if (insn->exceptionEncountered()) {
+      // Exception; don't pass insn to completedLoads_
+      raiseException_(insn);
+      return;
+    }
```

Comment:
This may be answered in my previous comment, but if raise exception is just setting commit ready, why do we need to use a call-back function type approach?

Reply:
That logic may not be appropriate for all core types using the LSQ (e.g. inorder vs OoO core)